src/other/tables.jl

```diff
@@ -0,0 +1,11 @@
+using Tables
+
+Tables.schema(df::DataFrame) = NamedTuple{Tuple(names(df)), Tuple{eltypes(df)...}}
+Tables.AccessStyle(df::DataFrame) = Tables.ColumnAccess()
+Tables.rows(df::DataFrame) = eachrow(df)
+Tables.columns(df::DataFrame) = df
+
+DataFrame(x::Tables.RowTable) = DataFrame([collect(u) for u in Tables.columns(x)],
+                                          collect(Tables.names(Tables.schema(x))))
+DataFrame(x::Any) = DataFrame([collect(u) for u in Tables.columns(x)],
```

Comment:
Can you please explain the rationale behind this catchall definition? Why it would be required (a general type might not support Tables.jl interface)?

Reply:
OTOH I would add `Tables.ColumnTable` constructor.

Reply:
I agree this duck-typing doesn't sound ideal. Would there be a trait to ensure we only try to do this on structures for which it makes sense? Otherwise I'm afraid it will throw confusing errors for users.

Reply:
That is what I am trying to convince to @quinnj here: https://github.com/JuliaData/Tables.jl/issues/5.

Reply:
This is really meant to replace the type-piracy currently in [IterableTables.jl](https://github.com/queryverse/IterableTables.jl/blob/45f95e18023058410434f21ed3faa7db66364f3d/src/integrations/dataframes.jl#L20) that defines it's own DataFrames catchall. I think we can come up w/ a simple trait to make this all work nicely too.